contributors/design-proposals/apiserver-build-in-admission-plugins.md

```diff
@@ -0,0 +1,31 @@
+# Build some Admission Controllers into the Generic API server library
+
+**Related PR:**
+
+| Topic | Link |
+| ----- | ---- |
+| Admission Control | https://github.com/kubernetes/community/blob/master/contributors/design-proposals/admission_control.md |
+
+## Introduction
+
+At the moment admission controllers are implemented as plugins and they have to be compiled into the final binary in order to be used at a later time. To streamline this process especially for aggregated API servers we would like to build some plugins into the generic API server library.
+
+## Implementation
+
+The first step would be to move some plugins from [admission pkg](https://github.com/kubernetes/kubernetes/tree/master/plugin/pkg/admission) to the generic API server library. We can start off by moving the namespace lifcycle controller. 
+
+The second step would involve creating a "standard" plugin initializer that would be part of the generic API server. It would use kubeconfig to populate [ClientSet](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubeapiserver/admission/initializer.go#L29) and [SharedInformers](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubeapiserver/admission/initializer.go#L35). By default for servers that would be run on the kubernetes cluster in-cluster config would be used. The standard initializer would also accept optional kubeconfig parameter that could be used by an aggregated API servers that are not part of the cluster. In that case the config can be easily populated form the file. 
+
+*Note*:
+For loading in-cluster configuration [visit](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/examples/in-cluster/main.go#L30)
+ For loading the configuartion directly from a file [visit](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/examples/out-of-cluster/main.go)
+ 
+## How to add a buildin admission plugin ?
+ At this point adding an admission plugin is very simple and boils down to performing the following series of steps:
```

Comment:
I think once you're done, the steps become, 

1. write admission plugin
2. register the plugin
3. reference the plugin in the admission chain (we should either support the argument or make manually writing the chain easy.  Maybe both).